<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Split data | WG2: Cell-type classification</title>
<meta name="author" content="Jose Alquicira Hernandez and Lieke Michelsen">
<meta name="description" content="For data integration and cell type classification purposes, it is recommended to split the data into smaller subsets that reflects a single homogeneous experimental unit. In the context of cell...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="Chapter 4 Split data | WG2: Cell-type classification">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/powellgenomicslab/WG2-pipeline-classification-docs/split-data.html">
<meta property="og:description" content="For data integration and cell type classification purposes, it is recommended to split the data into smaller subsets that reflects a single homogeneous experimental unit. In the context of cell...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Split data | WG2: Cell-type classification">
<meta name="twitter:description" content="For data integration and cell type classification purposes, it is recommended to split the data into smaller subsets that reflects a single homogeneous experimental unit. In the context of cell...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">WG2: Cell-type classification</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="general.html"><span class="header-section-number">2</span> General</a></li>
<li><a class="" href="getting-started.html"><span class="header-section-number">3</span> Getting started</a></li>
<li><a class="active" href="split-data.html"><span class="header-section-number">4</span> Split data</a></li>
<li><a class="" href="azimuth-classification.html"><span class="header-section-number">5</span> Azimuth classification</a></li>
<li><a class="" href="hierarchical-scpred-classification.html"><span class="header-section-number">6</span> Hierarchical scPred classification</a></li>
<li><a class="" href="merge-data.html"><span class="header-section-number">7</span> Merge data</a></li>
<li><a class="" href="compare-classifications.html"><span class="header-section-number">8</span> Compare classifications</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/powellgenomicslab/WG2-pipeline-classification-docs">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="split-data" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Split data<a class="anchor" aria-label="anchor" href="#split-data"><i class="fas fa-link"></i></a>
</h1>
<p>For data integration and cell type classification purposes, it is recommended to
split the data into smaller subsets that reflects a single homogeneous experimental
unit. In the context of cell type classification we define an <em>experimental unit</em> as
the group of cells sharing similar biological and technical variance components.
A batch (e.g pool/well from a single cell experiment) represents the most common
experimental unit.</p>
<p>Therefore, we recommend splitting the data for:</p>
<ol style="list-style-type: decimal">
<li>
<em>Faster classification/integration</em>. Splitting the data avoids extensive search
for cell anchors and neighbors in the azimuth approach</li>
<li>
<em>Parallelization</em>. Each experimental unit can be processed and analyzed independently
using multiple CPU workers</li>
<li>
<em>Avoiding intra-batch effects in the query dataset</em>. Cell differences driven by
batch effects within the query dataset may influence the performance of the integration/alignment.</li>
</ol>
<p>We recommend splitting the data by experimental units (e.g. batches). If the Seurat
query object represents a single experimental unit, no splitting is necessary and
this section can be skipped.</p>
<p>To split the data, we run the following code. First, let’s create an output directory</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="split-data.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> step1_split</span></code></pre></div>
<p>Let’s assume we have a Seurat RDS object called <code>query.RDS</code> with a metadata column name
called <code>pool</code> containing the values <code>1</code> and <code>2</code> corresponding to two independent pools. We can split the data by pool using the <code>split.R</code> module.</p>
<p>Let’s examine the parameter information:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="split-data.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">singularity</span> exec <span class="at">-B</span> <span class="va">$PWD</span> cell_classification.sif <span class="dt">\ </span></span>
<span id="cb4-2"><a href="split-data.html#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="ex">Rscript</span> split.R <span class="at">--help</span></span></code></pre></div>
<pre><code>Usage: split.R [options]


Options:
        --file=CHARACTER
                RDS object file name

        --batch=CHARACTER
                Batch column to split the data

        --out=CHARACTER
                Output file name [default= split_object]

        --path=CHARACTER
                Output path to store results [default= .]

        -h, --help
                Show this help message and exit
</code></pre>
<p>We can now split the data as follows</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="split-data.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">singularity</span> exec <span class="at">-B</span> <span class="va">$PWD</span> cell_classification.sif <span class="dt">\</span></span>
<span id="cb6-2"><a href="split-data.html#cb6-2" aria-hidden="true" tabindex="-1"></a>  Rscript /split.R <span class="dt">\</span></span>
<span id="cb6-3"><a href="split-data.html#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--file</span> query.RDS <span class="dt">\</span></span>
<span id="cb6-4"><a href="split-data.html#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--batch</span> pool <span class="dt">\</span></span>
<span id="cb6-5"><a href="split-data.html#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> query <span class="dt">\</span></span>
<span id="cb6-6"><a href="split-data.html#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--path</span> step1_split</span></code></pre></div>
<p><strong><code>split_object.R</code></strong> takes two main arguments. <code>--file</code> corresponds to the seurat object
filename in <code>.RDS</code> format. <code>--batch</code> expects a string corresponding to a column name
in the <code>@metadata</code> slot. A group of RDS files following the *{out}_{batch levels}.RDS*
filename pattern are created. In this example, two files called <code>query_1.RDS</code> and <code>query_2.RDS</code>
are stored in the <code>step1_split</code> directory.</p>
<blockquote>
<p>We strongly recommend storing results in different directories to avoid file
overwriting.</p>
</blockquote>

</div>
  <div class="chapter-nav">
<div class="prev"><a href="getting-started.html"><span class="header-section-number">3</span> Getting started</a></div>
<div class="next"><a href="azimuth-classification.html"><span class="header-section-number">5</span> Azimuth classification</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#split-data"><span class="header-section-number">4</span> Split data</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/powellgenomicslab/WG2-pipeline-classification-docs/blob/master/04-split_data.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/powellgenomicslab/WG2-pipeline-classification-docs/edit/master/04-split_data.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>WG2: Cell-type classification</strong>" was written by Jose Alquicira Hernandez and Lieke Michelsen. It was last built on 2021-11-24.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
